### PR TITLE
Use ok_or_else() everywhere else rather than handrolling

### PR DESCRIPTION
**Pull request by nabijaczleweli**
I fix things that bother me. This bothered me so I fixed it.

### CODE REVIEW

**Comment by dtolnay** [comment]
Thanks! I don't like some of these because they require horizontally scrolling the example code to read the line. Could you update to use a maximum of 80 characters per line? (I'll even let you use 84 if needed because that seems to fit in all the browsers I tried.)

File: _src/deserialize-struct.md [hunk 1]
```diff
@@ -92,18 +92,8 @@ impl<'de> Deserialize<'de> for Duration {
...
-                        return Err(de::Error::invalid_length(0, &self));
-                    }
-                };
-                let nanos: u32 = match seq.next_element()? {
-                    Some(value) => value,
-                    None => {
-                        return Err(de::Error::invalid_length(1, &self));
-                    }
-                };
+                let secs = match seq.next_element().ok_or_else(|| de::Error::invalid_length(0, &self))?;
```
Comment (dtolnay) [inline _src/deserialize-struct.md]: "match" should not be here.

Comment (nabijaczleweli) [inline _src/deserialize-struct.md]: I'm blind :v Fixed.

**Comment by nabijaczleweli** [comment]
Fixed the `match`es I missed and reduced to max 84CPL

**Comment by dtolnay** [comment]
Please rebase this on ee8837cd27a823f98763380d70e0aeecd56a9df9 so we get Travis to compile the code samples.

**Comment by nabijaczleweli** [comment]
Done, and it compiles! *audible cheering*

**Pull request merged**